### PR TITLE
Go reversely.

### PR DESCRIPTION
**Pull request by CPWstatic**
Pr #1090 which is submitted by @dutor totally meet our requirements. But we found that there are two things that need to be improved.
1. We need to save the actual data type when we get data from EdgeHolder.
2. We need to consider the _**ranking**_ when we fetch edge props in reversely traversal.

### CODE REVIEW

**Comment by CPWstatic** [comment]
Jenkins go!

File: src/graph/GoExecutor.cpp [hunk 1]
```diff
@@ -859,34 +1008,53 @@ bool GoExecutor::processFinalResult(RpcResponse &rpcResp, Callback cb) const {
...
 
-                        if (saveTypeFlag) {
-                            colTypes.back() = iter->getSchema()->getFieldType(prop).type;
-                        }
-                        if (edgeType != edgeStatus.value()) {
-                            auto sit = edgeSchema.find(edgeStatus.value());
-                            if (sit == edgeSchema.end()) {
-                                return Status::Error("get schema failed");
+                        if (isReversely()) {
+                            auto dst = RowReader::getPropByName(&*iter, "_dst");
```
Comment (laura-ding) [inline src/graph/GoExecutor.cpp]: modify to `_DST`

File: src/graph/GoExecutor.cpp [hunk 2]
```diff
@@ -1076,6 +1244,79 @@ void GoExecutor::VertexHolder::add(const storage::cpp2::QueryResponse &resp) {
...
+    auto result = RowReader::getPropByName(reader.get(), prop);
+    if (!ok(result)) {
+        return Status::Error("Prop not found: `%s'", prop.c_str());
+    }
+    return value(result);
+}
+
+
+SupportedType GoExecutor::EdgeHolder::getType(VertexID src,
+                                           VertexID dst,
```
Comment (laura-ding) [inline src/graph/GoExecutor.cpp]: alignment

File: src/graph/GoExecutor.cpp [hunk 3]
```diff
@@ -1076,6 +1244,79 @@ void GoExecutor::VertexHolder::add(const storage::cpp2::QueryResponse &resp) {
...
+            resp.get_data()->empty()) {
+        return;
+    }
+
+    auto eschema = std::make_shared<ResultSchemaProvider>(*resp.get_schema());
+    RowSetReader rsReader(eschema, *resp.get_data());
+    auto collector = std::make_unique<Collector>();
+    auto iter = rsReader.begin();
+    while (iter) {
+        auto src = collector->getProp(eschema.get(), "_src", &*iter);
```
Comment (laura-ding) [inline src/graph/GoExecutor.cpp]: modify to `_SRC` `_DST` `_TYPE`

File: src/graph/GoExecutor.cpp [hunk 4]
```diff
@@ -1076,6 +1244,79 @@ void GoExecutor::VertexHolder::add(const storage::cpp2::QueryResponse &resp) {
...
+        }
+        EdgeKey key;
+        std::get<0>(key) = boost::get<int64_t>(src.value());
+        std::get<1>(key) = boost::get<int64_t>(dst.value());
+        std::get<2>(key) = boost::get<int64_t>(type.value());
+        RowWriter rWriter(eschema);
+        auto fields = iter->numFields();
+        for (auto i = 0; i < fields; i++) {
+            auto result = RowReader::getPropByIndex(&*iter, i);
+            CHECK(ok(result));
```
Comment (laura-ding) [inline src/graph/GoExecutor.cpp]: delete `CHECK`

**Pull request merged**